lib/elixir/lib/uri.ex

```diff
@@ -204,9 +208,9 @@ defmodule URI do
       true
 
   """
-  @spec char_reserved?(term) :: boolean
-  def char_reserved?(c) do
-    c in ':/?#[]@!$&\'()*+,;='
+  @spec char_reserved?(pos_integer) :: boolean
+  def char_reserved?(char) when is_integer(char) and char > 0 do
```

Comment:
This (and similar clauses for `char_unreserved?/1` and `char_unescaped?/1`) technically break backwards compatibility I guess, but accepting anything was not a defined behaviour I think, so it should still be ok. Thoughts?